src/components/atoms/ButtonNG/ButtonNG.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { CSSProperties } from "react";
+import { Link } from "react-router-dom";
+import classNames from "classnames";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { faCircleNotch } from "@fortawesome/free-solid-svg-icons/faCircleNotch";
+
+import "./ButtonNG.scss";
+
+export type ButtonGradientType =
+  | "gradient"
+  | "admin-gradient"
+  | "danger-gradient";
+
+export type ButtonType = "button" | "reset" | "submit";
+export type ButtonVariant = "primary" | "secondary" | "white" | "dark";
+export type ButtonIconSize = "1x" | "2x" | "3x";
+
+export interface ButtonProps {
+  className?: string;
+  style?: CSSProperties;
+  disabled?: boolean;
+  onClick?: () => void;
+  isLink?: boolean;
+  linkTo?: string;
+  newTab?: boolean;
+  loading?: boolean;
+  gradient?: ButtonGradientType;
+  type?: ButtonType;
+  variant?: ButtonVariant;
+  iconOnly?: boolean;
+  iconName?: IconProp;
+  iconSize?: ButtonIconSize;
+}
+
+export const ButtonNG: React.FC<ButtonProps> = ({
+  children,
+  className = "",
+  style,
+  loading,
+  onClick,
+  type = "button",
+  gradient = "",
+  isLink = false,
+  linkTo = "",
+  disabled = false,
+  newTab = false,
+  variant = "",
+  iconOnly = false,
+  iconName,
+  iconSize = "1x",
+}) => {
+  const parentClasses = classNames({
+    "ButtonNG ButtonNG__link": isLink,
+    "ButtonNG ButtonNG__button": !isLink,
+    "ButtonNG--disabled": disabled,
+    "ButtonNG--enabled": !disabled,
+    "ButtonNG--loading": loading,
+    [`ButtonNG--icon-only ButtonNG--${iconSize}`]: iconOnly,
+    [`ButtonNG--icon-text`]: !iconOnly,
+    [`ButtonNG--${gradient}`]: gradient && !disabled,
+    [`ButtonNG--${variant}`]: variant && !disabled,
+    [className]: className,
+  });
+
+  const iconClasses = classNames({
+    "ButtonNG__icon ButtonNG__icon--icon-only": iconOnly,
+    "ButtonNG__icon ButtonNG__icon--icon-text": !iconOnly,
+  });
+
+  if (loading) {
+    return (
+      <button className={parentClasses} style={style} type={type}>
+        <FontAwesomeIcon
+          icon={faCircleNotch}
+          spin
+          size={iconSize}
+          className="ButtonNG__icon"
+        />
+      </button>
+    );
+  }
+
+  if (isLink) {
+    return (
+      <Link
```

Comment:
I opted to keep a minimal viable button that can be a drop-in replacement, so everything is neatly tucked into one tsx (+scss) file until there's a need for extra complexity by the demands of specific uses.

In any case, `ButtonNG` should serve as a façade if new specialized button components stem out of it and used locally only by it and not externally by all other components. This can ensure consistency as an outside component API and also reduce complexity by having the button façade as a switchboard for the few specialized buttons.

You don't need to "know" they're both used together because they're also both used apart.

Reply:
Example for the flags:
- `isLink` - signals form i.e. "display as a link"
- `linkTo` - signals function i.e. "where do I point to" 

It does look a bit confusing, but some of these are added just for parity with existing `Button` to facilitate easier transition.

I tried a shortcut of the form `const isLink = !! linkTo` but that might clash with a disabled link where `  to={disabled ? "#" : linkTo}`

Reply:
I'd leave them as they are now and figure out a possible refactor after all usages are known.